### PR TITLE
feat: Align with the new design in form for inline error display

### LINKED ISSUE #1623: [Linting and Validation] Align with the new design in form for inline error display
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
https://www.figma.com/file/Ux0vucOvaLL2xH22WpFgV65X/MarcB_Composer-UI?node-id=3278%3A26678

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by alanlong9278**
## Description

Align with the new design in form for inline error display

### Discussion
- Show detail error message.(Now the error message is fixed)
- Notification level. (Now only show error message, but notification message has 2 levels in notification page: warninig & error).

## Task Item

closes #1623 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code refactor (non-breaking change which improve code quality, clean up, add tests, etc)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Doc update (document update)

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have functionally tested my change

## Screenshots

validation

### CODE REVIEW

File: Composer/packages/extensions/obiformeditor/src/Form/fields/PromptField/PromptSettings.tsx [hunk 1]
```diff
@@ -19,52 +28,77 @@ interface PromptSettingsrops extends FieldProps<MicrosoftInputDialog> {
...
+            schema={getSchema('defaultValue')}
+            id={idSchema.defaultValue.__id}
+            value={formData.defaultValue}
+            label={formatMessage('Default value')}
+            formContext={props.formContext}
+            rawErrors={errorSchema.defaultValue && errorSchema.defaultValue.__errors}
+            hiddenErrMessage={true}
+            onValidate={onValidate}
+          />
+        </div>
```
Comment (a-b-r-o-w-n) [inline Composer/packages/extensions/obiformeditor/src/Form/fields/PromptField/PromptSettings.tsx]: Let's create a new form field that renders two schema fields and handles the error presentation.

File: Composer/packages/extensions/obiformeditor/src/Form/widgets/ExpressionWidget.tsx [hunk 2]
```diff
@@ -36,34 +24,109 @@ interface ExpresionWidgetProps extends ITextFieldProps {
...
+            rel="noopener noreferrer"
+          >
+            here
+          </Link>
+          .
+        </>
+      );
+      setErrInputStyle({
+        fieldGroup: {
+          borderColor: `${SharedColors.red20} !important`,
```
Comment (a-b-r-o-w-n) [inline Composer/packages/extensions/obiformeditor/src/Form/widgets/ExpressionWidget.tsx]: We shouldn't need to set the border color if we use `onGetErrorMessage`

**Comment by a-b-r-o-w-n** [comment]
Also, I didn't see anything to handle lint warnings. The spec is that we display an error if present, then a warning.

File: Composer/packages/extensions/obiformeditor/src/Form/widgets/ExpressionWidget.tsx [hunk 3]
```diff
@@ -26,36 +36,37 @@ const getErrorMessage = () =>
...
-  editable?: boolean;
-}
-
 export const ExpressionWidget: React.FC<ExpresionWidgetProps> = props => {
   const { rawErrors, formContext, schema, id, label, editable, ...rest } = props;
   const { shellApi } = formContext;
   const { description } = schema;
+  const onGetErrorMessage = (value: string): Promise<string | JSX.Element> => {
+    return new Promise(async resolve => {
+      const isValid = await shellApi.validateExpression(value);
```
Comment (a-b-r-o-w-n) [inline Composer/packages/extensions/obiformeditor/src/Form/widgets/ExpressionWidget.tsx]: This should happen after the null value check.

File: Composer/packages/extensions/obiformeditor/src/Form/widgets/ExpressionWidget.tsx [hunk 4]
```diff
@@ -26,36 +36,37 @@ const getErrorMessage = () =>
...
 
-    return '';
+      if (rawErrors && rawErrors.length > 0) {
+        errMessage = <>{rawErrors[0]}</>;
+      }
+      resolve(
+        <MessageBar
+          messageBarType={MessageBarType.error}
+          isMultiline={false}
+          dismissButtonAriaLabel="Close"
```
Comment (a-b-r-o-w-n) [inline Composer/packages/extensions/obiformeditor/src/Form/widgets/ExpressionWidget.tsx]: needs to be formatMessage

File: Composer/packages/extensions/obiformeditor/src/Form/widgets/ExpressionWidget.tsx [hunk 5]
```diff
@@ -26,36 +36,37 @@ const getErrorMessage = () =>
...
+          truncated={true}
+          overflowButtonAriaLabel="See more"
```
Comment (a-b-r-o-w-n) [inline Composer/packages/extensions/obiformeditor/src/Form/widgets/ExpressionWidget.tsx]: needs to be formatMessage

File: Composer/packages/lib/code-editor/src/RichEditor.tsx [hunk 6]
```diff
@@ -67,9 +67,15 @@ export function RichEditor(props: RichEditorProps) {
         <BaseEditor {...rest} placeholder={hidePlaceholder ? undefined : placeholder} />
       </div>
       {isInvalid && (
-        <div style={{ fontSize: '14px', color: SharedColors.red20 }}>
-          <span>{errorHelp}</span>
-        </div>
+        <MessageBar
+          messageBarType={MessageBarType.error}
+          isMultiline={false}
+          dismissButtonAriaLabel="Close"
```
Comment (a-b-r-o-w-n) [inline Composer/packages/lib/code-editor/src/RichEditor.tsx]: needs to be formatMessage

File: Composer/packages/lib/code-editor/src/RichEditor.tsx [hunk 7]
```diff
@@ -67,9 +67,15 @@ export function RichEditor(props: RichEditorProps) {
...
+          truncated={true}
+          overflowButtonAriaLabel="See more"
```
Comment (a-b-r-o-w-n) [inline Composer/packages/lib/code-editor/src/RichEditor.tsx]: needs to be formatMessage

File: Composer/packages/extensions/obiformeditor/src/Form/fields/PromptField/PromptSettings.tsx [hunk 8]
```diff
@@ -19,52 +28,77 @@ interface PromptSettingsrops extends FieldProps<MicrosoftInputDialog> {
...
-        <TextWidget
-          onChange={onChange('maxTurnCount')}
-          schema={getSchema('maxTurnCount')}
-          id={idSchema.maxTurnCount.__id}
-          value={formData.maxTurnCount}
-          label={formatMessage('Max turn count')}
-          formContext={props.formContext}
-          rawErrors={errorSchema.maxTurnCount && errorSchema.maxTurnCount.__errors}
-        />
+    <div>
```
[the remark this answers is not in the document]
Comment (a-b-r-o-w-n) [reply Composer/packages/extensions/obiformeditor/src/Form/fields/PromptField/PromptSettings.tsx]: Why is this resolved? Is the div necessary for style purposes?

[the remark this answers is not in the document]
Comment (alanlong9278) [reply Composer/packages/extensions/obiformeditor/src/Form/fields/PromptField/PromptSettings.tsx]: I'll create new field for two schema fields.

File: Composer/packages/extensions/obiformeditor/src/Form/ObjectFieldTemplate/styles.css [hunk 9]
```diff
@@ -4,6 +4,7 @@
 
 .ObjectItem .ObjectItemField {
   flex: 1;
+  overflow: hidden;
```
[the remark this answers is not in the document]
Comment (alanlong9278) [reply Composer/packages/extensions/obiformeditor/src/Form/ObjectFieldTemplate/styles.css]: uierror
Like the gif show, if we don't set `overflow: hidden`, the editor's width & errMessage width will be enlarged.

**Comment by a-b-r-o-w-n** [comment]
Note: What this change _does not_ do is surface errors for any arbitrary field in the form. Only LG, LU and expression errors are surfaced right now.

**Pull request merged**